### PR TITLE
feat(moo-ds): render tooltips and popovers in the top layer

### PR DESCRIPTION
**Pull request by AndrewMcLachlan**
Groundwork for #657 Tier 1.2 — and the reason that workstream **can't start with the modal**.

> **Stacked on #682** (→ #681). Merge order: #681 → #682 → this.

## Why this has to come first

The overlay z-index ladder deliberately puts tooltips (1080) and popovers (1070) **above** the modal (1055). The native top layer ignores `z-index` entirely — measured, an open dialog paints above `z-index: 2147483647`.

So promoting `Modal` to `<dialog>` while tooltips still rely on z-index would **bury every tooltip and popover opened inside a modal**. Moving them first means that ordering already holds when the modal migration lands.

Verified against a real `<dialog>`: with the trigger inside an open modal, the tooltip paints over the modal's content.

It's also a win today independent of any modal work — both overlays become immune to an ancestor stacking context, which `z-index` alone never was.

## Feature detected, not applied unconditionally

A `[popover]` element is `display: none` until opened. Adding the attribute where `showPopover()` doesn't exist — jsdom, or a browser predating the API — **hides the overlay outright and drops it from the accessibility tree**.

That isn't hypothetical. Applying it unconditionally broke the Tooltip test immediately:

```
TestingLibraryElementError: Unable to find an accessible element with the role "tooltip"
```

`topLayerProps()` omits the attribute in those environments, so they keep the previous behaviour — which is why the `z-index` declarations stay as the fallback. Covered by 5 unit tests including the not-callable edge case.

`"manual"` rather than `"auto"`: auto adds light-dismiss and would close these on any outside click, fighting the components' own open/close state.

## UA styles undone

`[popover]` defaults to `inset: 0` with `margin: auto` — which centres the element and fights `position-area` — plus a border, padding, `Canvas` background and `overflow: auto` that would paint a box behind the content and clip the tooltip's arrow. All reset in CSS.

## A finding recorded for the modal work, not fixed here

`showModal()` makes everything outside the dialog **inert**. An overlay portalled to `document.body` is therefore painted above the modal but is **not hit-testable and leaves the accessibility tree**.

I caught this because my first hit-test showed the tooltip losing — the screenshot then showed it painting correctly on top, so the failure was inertness, not paint order.

For tooltips that's tolerable (they're read via `aria-describedby`), but an **interactive** overlay — the user menu, say — would stop working inside a modal. Those will need to portal into the dialog subtree. Noted in the commit so the modal work inherits it rather than rediscovering it.

## Verification

- `npm run build -w @andrewmclachlan/moo-ds` — clean
- `npm run test:run` — **1709/1709** pass (1704 + 5 new)
- Top-layer paint order confirmed by screenshot against a real `<dialog>`
- Anchor positioning, `position-try-fallbacks` and `@starting-style` all confirmed still working with the popover attribute applied

Refs #657

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: moo-ds/src/components/Tooltip.tsx [hunk 1]
```diff
@@ -17,6 +18,14 @@ export const Tooltip: React.FC<PropsWithChildren<{ id: string }>> = ({ id, child
 
     useLayoutEffect(() => {
         if (show && portalRef.current && triggerRef.current) {
+            // Promote into the top layer. The tooltip previously relied on
+            // z-index: 1080, which loses to anything on the page carrying a
+            // higher one -- a third-party widget, an app-level overlay --
+            // and loses outright to a top-layer dialog, which paints above
+            // any z-index at all. Optional call so environments without the
+            // popover API (jsdom) just render it in place as before.
+            portalRef.current.showPopover?.();
```
[the remark this answers is not in the document]
Comment (AndrewMcLachlan) [reply moo-ds/src/components/Tooltip.tsx]: Correct, and it contradicted my own predicate: `supportsPopover()` tests `typeof === "function"` and I even have a test for the not-callable case — so the attribute was withheld in exactly the situation where the call would still have been attempted.

Both call sites now go through a `showInTopLayer()` helper that guards on the element's own method, so the check that decides whether to apply the attribute and the check that decides whether to call the API can't drift apart.

Verified the tests actually bite: restoring the `?.()` form fails exactly the "not a function" case and leaves the other eight green. Fixed in 60de94c.

File: moo-ds/src/components/OverlayTrigger.tsx [hunk 2]
```diff
@@ -37,6 +38,13 @@ export const OverlayTrigger: React.FC<OverlayTriggerProps> = ({
     // _overlay.css rather than in inline styles.
     useLayoutEffect(() => {
         if (show && overlayRef.current) {
+            // Promote into the top layer -- see the note in Tooltip.tsx. The
+            // z-index: 1070 this replaces loses to any higher z-index on the
+            // page, and to a top-layer dialog outright. Optional call so
+            // jsdom, which has no popover API, keeps rendering the overlay in
+            // place.
+            overlayRef.current.showPopover?.();
```
[the remark this answers is not in the document]
Comment (AndrewMcLachlan) [reply moo-ds/src/components/OverlayTrigger.tsx]: Same fix — this call site now goes through the shared `showInTopLayer()` guard rather than `?.()`.

I also checked a neighbouring case while here: calling `showPopover()` on an already-open popover is a **no-op, not an error**, so a re-running effect needs no additional guard. Fixed in 60de94c.
